File: PDM(Programacao dispositivos moveis)/crud/app/src/main/java/cristian/crud/MainActivity.java
package cristian.crud;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    AppCompatButton redirectBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        redirectBtn = (AppCompatButton) findViewById(R.id.redirectCreate);

        redirectBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(MainActivity.this, InserirActivity.class);
                startActivity(i);            }
        });
    }
    @Override
    protected void onResume(){
        super.onResume();
        loadData();
    }

    private void loadData() {
        DBHelper db = new DBHelper(getApplicationContext());
        List<String> lista = db.getContatos();
        ArrayAdapter<String> adaptador = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, lista);
        ListView lv = findViewById(R.id.listView);
        lv.setAdapter(adaptador);
    }


}

File: PDM(Programacao dispositivos moveis)/PDM2/MusicApp/app/src/main/java/com/cran/musicapp/MainActivity.kt
package com.cran.musicapp

import android.annotation.SuppressLint

import android.content.res.AssetFileDescriptor
import android.media.MediaPlayer
import android.os.Bundle

import android.util.Log
import android.view.View

import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton

import androidx.core.content.ContextCompat
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.chibde.visualizer.SquareBarVisualizer


import java.io.IOException

class MainActivity : AppCompatActivity() {

    lateinit var mediaPlayer: MediaPlayer
    lateinit var visualizer: SquareBarVisualizer

    @SuppressLint("MissingInflatedId")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        mediaPlayer = MediaPlayer()
        try {
            val assetDescriptor: AssetFileDescriptor = assets.openFd("eminem_superman.mp3")
            mediaPlayer.setDataSource(
                assetDescriptor.fileDescriptor, assetDescriptor.startOffset, assetDescriptor.length
            )
            assetDescriptor.close()
            mediaPlayer.prepare()
        } catch (e: IOException) {
            Log.e("MainActivity", "Error setting data source: ${e.message}")

        }

        val playButton: AppCompatButton = findViewById(R.id.btnPlay)
        playButton.setOnClickListener { play(it) }


    }

    fun play(view: View) {
        try {
            mediaPlayer.setVolume(1f, 1f);

            mediaPlayer.start()
            visualizer.setGap(2)
            visualizer.setDensity(256F)
            visualizer.setColor(R.color.white)
            visualizer.setPlayer(mediaPlayer.audioSessionId)


        } catch (e: Exception) {
            Log.e("IFTM", e.message.toString())
            Log.e("IFTM", e.toString())
        }
    }


}